themes/default/content/blog/kubernetes-fundamentals-part-two/index.md

```diff
@@ -0,0 +1,215 @@
+<!--more-->
+
+## What is a managed Kubernetes service?
+
+Essentially, this just means that someone else is responsible for some of the work involved in standing up and maintaining Kubernetes. It varies from provider to provider, but services can be pretty bare-bones or include everything under the sun, handing you a pre-configured environment and a dedicated support team to help you.
+
+## What makes this useful?
+
+Kubernetes is very powerful, but also very complicated. The flexibility of the platform means that there's a lot of configuration you have to do just to get off the ground, and a managed service takes some of that off your plate. Some configurations are abstracted away, and some decisions are made for you. Things you would otherwise have to set up yourself, like load balancing, monitoring, and updates are covered by the vendor. It's less effort to get started, and less effort to maintain. Scaling is painless, since you can just provision more resources. For most people, a managed service is flat-out more reliable.
+
+
+## Okay, I want to try it for myself.
+
+First things first, we need [Pulumi](https://www.pulumi.com/docs/get-started/install/), [Node.js](https://nodejs.org/en/download/), and [npm](https://www.npmjs.com/get-npm) installed.
+
+We're going to do this on GCP, so you'll also need the [Google Cloud SDK](https://cloud.google.com/sdk/docs/downloads-interactive) set up. From there, log in with `gcloud`:
+
+```
+$ gcloud auth login
+$ gcloud config set project <YOUR_GCP_PROJECT_HERE>
+$ gcloud auth application-default login
+```
+
+## Creating a Cluster with Google Kubernetes Engine
+
+Next, we need a cluster. Start by creating a new directory, navigating into it, and initializing a new Pulumi project using typescript.
+
+```bash
+mkdir gke-hello-world && cd gke-hello-world && pulumi new typescript
+```
+
+We also need to install some dependencies, and set some required configuration values.
+
+```bash
+npm install --save @pulumi/pulumi @pulumi/gcp @pulumi/kubernetes
+pulumi config set gcp:project <YOUR_GCP_PROJECT_HERE>
+pulumi config set gcp:zone us-west1-a     // any valid GCP Zone here
+```
+
+Now, in `index.ts` add the following code:
+
+```typescript
+import * as k8s from "@pulumi/kubernetes";
+import * as pulumi from "@pulumi/pulumi";
+import * as gcp from "@pulumi/gcp";
+
+const name = "helloworld";
+
+// Create a GKE cluster
+const engineVersion = gcp.container.getEngineVersions().then(v => v.latestMasterVersion);
+const cluster = new gcp.container.Cluster(name, {
+    initialNodeCount: 2,
```

Comment:
Since the audience would be devs, do you think it would be worthwhile to explain why we might want 2 nodes and n1 standard machines?

Reply:
Mmmm, yes it's probably helpful enough. I'll add that too.